code/_onclick/ai.dm

```diff
@@ -20,6 +20,13 @@
 		return
 	next_click = world.time + 1
 
+	var/list/modifiers = params2list(params)
+
+	var/obj/vehicle/sealed/mecha/mech = loc //I know. Shaddup.
+	if(istype(mech) && (!mech.weapons_safety || LAZYACCESS(modifiers, MIDDLE_CLICK))) //mech and (safeties off or middle click)
+		SEND_SIGNAL(src, COMSIG_MOB_CLICKON, A, modifiers)
+		return
```

Comment:
blah blah ai code overrides clickon without calling parent. Just send the signal straight up instead of this mech snowflake check